pycaret/internal/preprocess/time_series/forecasting/preprocessor.py

```diff
@@ -119,7 +119,9 @@ def _transformation(
         ):
             self._add_transformation_steps(transform=transform_exogenous, target=False)
 
-    def _add_transformation_steps(self, transform: str, target: bool = True):
+    def _add_transformation_steps(
+        self, transform: Optional[Union[str, int, float]], target: bool = True
```

Comment:
I do not think the types are correct here. "transform" has to be a string. Why do you think it should include int and float and why is it Optional?

Reply:
Sorry, the type transform may not have been `Optional`. The process of calling `_add_transformation_steps` is as follows. The value entered here as transform is `transform_target` or `transform_exogenous`. Type `transform_target` and `transform_exogenous` are `Optional[Union[str, int, float]]`. Therefore, I defined the type `Optional[Union[str, int, float]]` for `transform`.

```python
    def _transformation(
        self,
        transform_target: Optional[Union[str, int, float]],
        transform_exogenous: Optional[Union[str, int, float]],
        exogenous_present: bool,
    ):
        # Impute target ----
        if transform_target is not None:
            self._add_transformation_steps(transform=transform_target, target=True)

        # Impute Exogenous ----
        # Only add exogenous pipeline steps if exogenous variables are present.
        if (
            exogenous_present == TSExogenousPresent.YES
            and transform_exogenous is not None
        ):
            self._add_transformation_steps(transform=transform_exogenous, target=False)
```

Also, the following types are being checked in the function named `_add_transformation_steps` and need to be corrected as I suggested. Alternatively, it would be preferable to check the types outside of the function named `_add_transformation_steps`.

https://github.com/pycaret/pycaret/blob/6d1e192103560250862b63f39e065c35b1c8bc2b/pycaret/internal/preprocess/time_series/forecasting/preprocessor.py#L151

Reply:
> Sorry, the type transform may not have been Optional. The process of calling _add_transformation_steps is as follows. The value entered here as transform is transform_target or transform_exogenous. Type transform_target and transform_exogenous are Optional[Union[str, int, float]]. Therefore, I defined the type Optional[Union[str, int, float]] for transform

But there is a check for None before calling the function so the input to `_add_transformation_steps` should not be Optional.

https://github.com/pycaret/pycaret/blob/a8d7f0def72030e7562898a40bd001768c790a42/pycaret/internal/preprocess/time_series/forecasting/preprocessor.py#L111

> Also, the following types are being checked in the function named _add_transformation_steps and need to be corrected as I suggested. Alternatively, it would be preferable to check the types outside of the function named _add_transformation_steps.

This function is called from the `setup` itself, so it should be fine. From a users flow perspective, they will be notified of the error immediately.

Reply:
@ngupta23 
Regarding this I have fix based on comments. 

https://github.com/pycaret/pycaret/pull/3081/commits/20dd505f5c6a4be3200526069ceac39f19367991

Thank you.